### PR TITLE
modify resetBlock to populate BlockHeader with the latestBlockHeader && add unittest for broadcastSignReq, checkAndSave and reportBlock

### PR DESCRIPTION
**Pull request by cyjseagull**
1.  modify resetBlock to populate BlockHeader with the latestBlockHeader
2. modify the implementation of resetSealingBlock to generate new block with latest blockheader
3. add unittest for broadcastSignReq, checkAndSave && reportBlock

### CODE REVIEW

File: libconsensus/pbft/PBFTConsensus.cpp [hunk 1]
```diff
@@ -536,10 +563,11 @@ void PBFTConsensus::handlePrepareMsg(PrepareReq& prepareReq, bool self)
     LOG(DEBUG) << "handlePrepareMsg, timecost=" << 1000 * t.elapsed();
 }
 
+///
 void PBFTConsensus::checkAndCommit()
 {
     u256 sign_size = m_reqCache->getSigCacheSize(m_reqCache->prepareCache().block_hash);
-    if (sign_size == minValidNodes())
+    if (sign_size >= minValidNodes())
```
Comment (wheatli) [inline libconsensus/pbft/PBFTConsensus.cpp]: The logic enable multiple times entrance,

File: libconsensus/pbft/PBFTConsensus.h [hunk 2]
```diff
@@ -454,14 +455,15 @@ class PBFTConsensus : public Consensus
...
+        /// get leader failed or this prepareReq is not broadcasted from leader
         if (!leader.first || req.idx != leader.second)
             return false;
         return true;
     }
 
     inline std::pair<bool, u256> getLeader() const
     {
-        if (m_cfgErr || m_leaderFailed || m_highestBlock.number() == 0)
+        if (m_cfgErr || m_leaderFailed || m_highestBlock.number() == INT64_MAX)
```
Comment (wheatli) [inline libconsensus/pbft/PBFTConsensus.h]: The default value of number is 0, any problem here?

File: libconsensus/pbft/PBFTConsensus.cpp (hunk 1 above)
Comment (cyjseagull) [reply libconsensus/pbft/PBFTConsensus.cpp]: has already been fixed, please refer to https://github.com/FISCO-BCOS/lab-bcos/pull/95/files#diff-7739c3316db232bb609da12413f4982eR573

File: libconsensus/pbft/PBFTConsensus.h (hunk 2 above)
Comment (cyjseagull) [reply libconsensus/pbft/PBFTConsensus.h]: has fixed the logic to determine a blockHeader is invalid by sealer()

please refer to: https://github.com/FISCO-BCOS/lab-bcos/pull/95/files#diff-4e84504bf7fa2ce1c95ef09ed00e5cfeR466

**Pull request merged**